core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -3206,29 +3206,11 @@ private boolean accept_(RexNode e, List<RexNode> newTerms) {
 
     private boolean accept2(RexNode left, RexNode right, SqlKind kind,
         List<RexNode> newTerms) {
-      switch (left.getKind()) {
-      case INPUT_REF:
-      case FIELD_ACCESS:
-      case CAST:
-        switch (right.getKind()) {
-        case LITERAL:
-          return accept2b(left, kind, (RexLiteral) right, newTerms);
-        default:
-          break;
-        }
-        return false;
-      case LITERAL:
-        switch (right.getKind()) {
-        case INPUT_REF:
-        case FIELD_ACCESS:
-        case CAST:
-          return accept2b(right, kind.reverse(), (RexLiteral) left, newTerms);
-        default:
-          break;
-        }
-        return false;
-      default:
-        break;
+      if (right.isA(SqlKind.LITERAL) && RexUtil.isDeterministic(left)) {
```

Comment:
None of these accept functions has any JavaDoc. I know they are private, but why not?
Not having a spec makes it harder to review code that uses or modifies them.
Maybe you can at least document this one and accept2b since you're touching them?